app/controllers/assignments_controller.rb

```diff
@@ -229,7 +229,6 @@ def create
     @assignment.transaction do
       begin
         @assignment = process_assignment_form(@assignment)
-        @assignment.token_start_date = @assignment.due_date
```

Comment:
As I mentioned above, I don't want `token_start_date` to be `nil` unnecessarily, so we should set a default value here; the appropriate thing to do is to change the default value.

You could instead use `Time.current` here, similar to what you were doing below.